src/arrayMethodJoin.js

```diff
@@ -5,7 +5,33 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = String(this[0]);
+    let separatorVariable = separator;
+
+    if (this[0] === undefined || this[0] === null) {
+      result = '';
+    }
+
+    if (this.length === 0) {
+      result = '';
+
+      return result;
+    }
+
+    if (separator === undefined) {
+      separatorVariable = ',';
+    }
+
+    const thisCopy = [...this];
+
+    for (let i = 1; i < this.length; i++) {
+      thisCopy[this.indexOf(undefined, i)] = '';
+      thisCopy[this.indexOf(null, i)] = '';
+
+      result += String(separatorVariable) + (thisCopy[i]);
+    }
```

Comment:
You can simplyfy your code, in for loop you can check if **this[i]** !== undefined oraz null and only then assign to **result**  value of **this[i]**, next you can assign **separator**, but you need to ommit this step for the last element
```suggestion
    for (let i = 1; i < this.length; i++) {
      //if (this[i] not equal to undefined and null) {
      //   result += this[i];
      //  }
      // if(condition when to add separator) {
      //  result += separator;
      // }
    }
```